server.go

```diff
@@ -1,10 +1,42 @@
 package dnscrypt
 
 import (
+	"net"
+	"sync"
+	"time"
+
+	"golang.org/x/net/context"
+
 	"github.com/AdguardTeam/golibs/log"
 	"github.com/miekg/dns"
 )
 
+// default read timeout for all reads
+const readTimeout = 2 * time.Second
+
+// in case of TCP we only use readTimeout for the first read
+// then we start using tcpIdleTimeout
+const tcpIdleTimeout = 8 * time.Second
```

Comment:
If this is a default value, I personally would also name it so, `defaultTCPIdleTimeout`.